tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedThreadFactory/resourcedef/ManagedThreadFactoryDefinitionBean.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package ee.jakarta.tck.concurrent.spec.ManagedThreadFactory.resourcedef;
+
+import jakarta.ejb.Local;
+import jakarta.ejb.Stateless;
+import jakarta.enterprise.concurrent.ContextServiceDefinition;
+import jakarta.enterprise.concurrent.ManagedThreadFactoryDefinition;
+
+import static jakarta.enterprise.concurrent.ContextServiceDefinition.APPLICATION;
+import static jakarta.enterprise.concurrent.ContextServiceDefinition.TRANSACTION;
+
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import ee.jakarta.tck.concurrent.common.context.IntContext;
+import ee.jakarta.tck.concurrent.common.context.StringContext;
+import ee.jakarta.tck.concurrent.spec.ContextService.contextPropagate.ContextServiceDefinitionServlet;
+
+/**
+ * @ContextServiceDefinitions are defined under {@link ContextServiceDefinitionServlet}
+ */
+@ManagedThreadFactoryDefinition(name = "java:app/concurrent/EJBThreadFactoryA",
+                                context = "java:app/concurrent/EJBContextD",
+                                priority = 4)
+@ManagedThreadFactoryDefinition(name = "java:comp/concurrent/EJBThreadFactoryB")
+//TODO: Can we use context from ContextServiceDefinitionBean?
```

Comment:
java:app/concurrent/EJBContextA could be used if it is in the same application due its java:app name. Leaving it this way is fine.  I'm pretty sure I saw at least one of your tests use a ContextService from a different EJB already, so you do have that coverage elsewhere.

Reply:
Fixed the issue was preventing loading from a different EJB, as the other EJB wasn't initialized first.